File: orchalanguage/src/main/java/orcha/lang/compiler/OrchaProgram.java
package orcha.lang.compiler;

import java.util.List;

public class OrchaProgram {

    List<IntegrationNode> integrationGraph;
    OrchaMetadata orchaMetadata;

    public OrchaProgram(){
    }

    public OrchaProgram(List<IntegrationNode> integrationGraph, OrchaMetadata orchaMetadata) {
        this.integrationGraph = integrationGraph;
        this.orchaMetadata = orchaMetadata;
    }

    public List<IntegrationNode> getIntegrationGraph() {
        return integrationGraph;
    }

    public void setIntegrationGraph(List<IntegrationNode> integrationGraph) {
        this.integrationGraph = integrationGraph;
    }

    public OrchaMetadata getOrchaMetadata() {
        return orchaMetadata;
    }

    public void setOrchaMetadata(OrchaMetadata orchaMetadata) {
        this.orchaMetadata = orchaMetadata;
    }
}

File: orchalang/src/main/kotlin/orcha/lang/compiler/OrchaProgram.kt
package orcha.lang.compiler

import orcha.lang.compiler.IntegrationNode
import orcha.lang.compiler.OrchaMetadata


data class OrchaProgram(var orchaMetadata: OrchaMetadata? = null, var integrationGraph: List<IntegrationNode>? = null) {

    constructor(): this(null, null) {}

}
